torch/overrides.py

```diff
@@ -1819,6 +1817,16 @@ def __init__(self):
     def __torch_function__(self, func, types, args=(), kwargs=None):
         raise NotImplementedError()
 
+    def __enter__(self):
+        if hasattr(self, "inner"):
+            raise RuntimeError(f"{self} has already been used as a mode, please create and use a fresh version")
+        old = _get_torch_function_mode()
```

Comment:
@ezyang In the old `push_torch_function_mode`, if old was None, we would make the inner be `BaseTorchFunctionMode` whose torch function just called the original function with the original args. Do you remember why?

Reply:
Yes it is so that self.inner.__torch_dispatch__() always worked. But since we strip off the original mode before calling into the function you can do func(*args, **kwargs) so supporting direct inner calls is much less pressing so I'm OK with dropping base mode (this is good for efficiency too, that base mode goops up stack frames).